src/test/scala/is/hail/distributedmatrix/BlockMatrixIsDistributedMatrixSuite.scala

```diff
@@ -0,0 +1,187 @@
+package is.hail.distributedmatrix
+
+import is.hail.check.Arbitrary._
+import is.hail.check.Prop._
+import is.hail.check.Gen._
+import is.hail.check._
+import is.hail.utils._
+import org.apache.spark.mllib.linalg._
+import org.apache.spark.mllib.linalg.distributed._
+
+import is.hail.SparkSuite
+import org.testng.annotations.Test
+
+class BlockMatrixIsDistributedMatrixSuite extends SparkSuite {
+  import is.hail.distributedmatrix.DistributedMatrix.implicits._
+
+  val dm = DistributedMatrix[BlockMatrix]
+  import dm.ops._
+
+  def toBM(rows: Seq[Array[Double]]): BlockMatrix =
+    new IndexedRowMatrix(sc.parallelize(rows.zipWithIndex.map { case (v, i) => IndexedRow(i, new DenseVector(v)) }),
+      rows.size, if (rows.isEmpty) 0 else rows.head.length)
+      .toBlockMatrixDense()
+
+  def toBM(rows: Seq[Array[Double]], blockSize: Int): BlockMatrix =
+    new IndexedRowMatrix(sc.parallelize(rows.zipWithIndex.map { case (v, i) => IndexedRow(i, new DenseVector(v)) }),
+      rows.size, if (rows.isEmpty) 0 else rows.head.length)
+      .toBlockMatrixDense()
+
+  val blockMatrixGenerator: Gen[BlockMatrix] = for {
+    (l, w) <- Gen.squareOfAreaAtMostSize
+    arrays <- Gen.buildableOfN[Seq, Array[Double]](l, Gen.buildableOfN(w, arbDouble.arbitrary))
+  } yield toBM(arrays)
```

Comment:
Should you gen the block size, too?  If the rows and cols per block are generated independently, that would have caught the diagonal bug.